File: scala/009/Main.scala
/*
 * ProjectEuler 009
 * Scala / Main.scala
 * copyright (c) 2015 Susisu
 *
 * Scala 2.11.4
 * `scalac Main.scala -d bin`
 */

object Main {
    def main(args: Array[String]) {
        for (a <- 1 to 332) {
            var b = a + 1;
            while (b < 1000 - a - b) {
                if (2 * a * b - 2000 * (a + b) + 1000000 == 0) {
                    println(a * b * (1000 - a - b));
                    return;
                }
                b += 1;
            }
        }
    }
}

File: rust/009/main.rs
/*
    ProjectEuler 009
    Rust / main.rs
    copyright (c) 2015 Susisu

    Rust 1.3.0
    `rustc main.rs -o bin/main`
*/

fn main() {
    for a in 1..334 {
        for b in (a + 1)..((1000 - a + 1) / 2) {
            let c: u32 = 1000 - a - b;
            if a.pow(2) + b.pow(2) == c.pow(2) {
                println!("{}", a * b * c);
                return;
            }
        }
    }
}
